File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
       Map<Integer, Integer> mp = new HashMap<Integer, Integer>();
       for( int ele: nums)
       {  
          if(mp.get(ele)==null)
          {
            mp.put(ele, 1);
          }
          else {
            return true;
            // mp.put(ele,mp.get(ele)+1);
          }
       } 

       return false;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int>mp;
        vector<int>ans;
        int i=0;
        for(auto ele: nums)
        {
            if(mp.count(target-ele))
            {   
                ans.push_back(mp[target-ele]);          
                ans.push_back(i);
                return ans;
            }
            else {
                mp[ele]=i;
            }

            i++;
        }
    }
};
